From 66b13628a3cb00f8c2f2a4613992145bf4fc17e3 Mon Sep 17 00:00:00 2001
From: Peng Fan <peng.fan@nxp.com>
Date: Sun, 30 Sep 2018 16:38:21 +0800
Subject: [PATCH 4789/5242] MLK-19728-6 ARM64: dts: imx8qm: passthrough
 isl29023 to domu

commit  479b42a611b2ac2a27e7b53973c62f485864e982 from
https://source.codeaurora.org/external/imx/linux-imx.git

Passthrough to domu.

 BuildInfo:
  - SCFW a27e9643, SECO-FW 31fabbff, IMX-MKIMAGE 7eae081f, ATF e6db8a6
  - U-Boot 2018.03-00007-gb8296e64d2

Signed-off-by: Peng Fan <peng.fan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qm-mek-dom0.dts     |    8 ++++++++
 .../boot/dts/freescale/fsl-imx8qm-mek-domu.dts     |   15 +++++++++++++++
 2 files changed, 23 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-dom0.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-dom0.dts
index 843f470..1ac3889 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-dom0.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-dom0.dts
@@ -184,11 +184,15 @@
 				SC_P_QSPI1A_SS0_B
 				SC_P_USB_SS3_TC3
 				SC_P_QSPI1A_DATA0
+				/* isl29023 */
+				SC_P_USDHC2_WP
+
 				>;
 			gpios = <&gpio1 13 GPIO_ACTIVE_LOW>,
 				<&gpio1 27 GPIO_ACTIVE_LOW>,
 				<&gpio4 6 GPIO_ACTIVE_LOW>,
 				<&gpio4 9 GPIO_ACTIVE_LOW>,
+				<&gpio4 11 GPIO_ACTIVE_HIGH>,
 				<&gpio4 19 GPIO_ACTIVE_HIGH>,
 				<&gpio4 26 GPIO_ACTIVE_HIGH>,
 				<&gpio4 27 GPIO_ACTIVE_LOW>,
@@ -615,6 +619,10 @@
 };
 
 &i2c0 {
+	isl29023@44 {
+		xen,passthrough;
+	};
+
 	fxos8700@1e {
 		xen,passthrough;
 	};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts
index cebc06e..07471bd 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts
@@ -795,6 +795,11 @@
 			>;
 		};
 
+		pinctrl_isl29023: isl29023grp {
+			fsl,pins = <
+				SC_P_USDHC2_WP_LSIO_GPIO4_IO11		0x00000021
+			>;
+		};
 	};
 };
 
@@ -858,6 +863,16 @@
 	clock-frequency = <100000>;
 	status = "okay";
 
+	isl29023@44 {
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_isl29023>;
+		compatible = "fsl,isl29023";
+		reg = <0x44>;
+		rext = <499>;
+		interrupt-parent = <&gpio4>;
+		interrupts = <11 2>;
+	};
+
 	fxos8700@1e {
 		compatible = "fsl,fxos8700";
 		reg = <0x1e>;
-- 
1.7.9.5

